### ISSUE TITLE
level 6 test case 3

### ISSUE THREAD

**Issue by theshaydays** [nus-cs2030/2122-s2#521]
# Summary
Hi I was wondering if anyone could help me figure out why my code isn't reading the input service times in the correct order. From what I see, the only error is that the first waiting customer in each server is reading each other's service time (i.e customer 4 is taking in customer 6's service time and vice versa). I swapped the two values in the test file and it works but idk why the code reads just these 2 customers in the swapped order. thanks!

## Screenshots (if any): 
![image](https://user-images.githubusercontent.com/100525532/164911882-f783a9c3-be92-41dc-bb87-4e2e77cf595e.png)

**Comment by ChoonLeng773** [comment]
same problem here m8

**Comment by pearlynliu** [comment]
hi! may i know in which event do you get the customer's service time? because i think the error might be that you called the service time too early. for me, i get() the service time in Serve event's execute. I am not sure if you did the same, but now it seems like u called the service time in the event before Serve

**Comment by vvarshh** [comment]
Hi! this is most likely because you called the service time too early. You should only be calling it in Serve. If not the wrong serving times can be assigned resulting in your output. Hope this helps!

**Comment by jlee1007** [comment]
how does ur implementation look like? 
only call the serve time at server and call it once per event! if you accidentally .get twice you would miss some numbers too!

**Comment by vvarshh** [comment]
Can I also check if you made an extra class called pending or smith of that idea ? might help with the implementation as well

**Comment by theshaydays** [comment]
@vvarshh nope i didn't which is quite concerning but it works so far. @jlee1007 the only other place i call serve time is at Wait but i don't call this customer's wait values but rather the other customer's serve values. The idea is that the serve event of a waiting customer is the sum of the serving customer's arrival time + the serving customers's serve time and all the serving time of all the other people waiting in front. My hope was that calling on the serve time of another customer  wouldn't be an issue because it's already been called in another customer object separate to the one at Wait

**Comment by pearlynliu** [comment]
> @vvarshh nope i didn't which is quite concerning but it works so far. @jlee1007 the only other place i call serve time is at Wait but i don't call this customer's wait values but rather the other customer's serve values. The idea is that the serve event of a waiting customer is the sum of the serving customer's arrival time + the serving customers's serve time and all the serving time of all the other people waiting in front. My hope was that calling on the serve time of another customer wouldn't be an issue because it's already been called in another customer object separate to the one at Wait

i agree with @jlee1007. i think your current implementation might be calling the get service time too many times. it should only be called in the Serve event because according to the description for level 5, the "the service time of the customer is not known until he/she is being served". if you are calling the service time of all the other people waiting in front, it might not be following this requirement. 

the Pending event as recommended by @vvarshh can also help you to check whether the customer is at the front of the queue. without Pending, i am not sure if there is another way to stall the Serve event of the customer without calling Wait event again.

**Comment by bl7447** [comment]
> @vvarshh nope i didn't which is quite concerning but it works so far. @jlee1007 the only other place i call serve time is at Wait but i don't call this customer's wait values but rather the other customer's serve values. The idea is that the serve event of a waiting customer is the sum of the serving customer's arrival time + the serving customers's serve time and all the serving time of all the other people waiting in front. My hope was that calling on the serve time of another customer wouldn't be an issue because it's already been called in another customer object separate to the one at Wait

I have this same issue and same logic, but I realised that with the project's rule of only knowing service time at the Serve event , it wouldn't work for test case 3 as when you have multiple people waiting for 1 server, to determine the waiting time for the 2nd waiting customer onwards, you would end up getting the service time of the customer in front too early before they have even been served yet, messing up the service time input order. So I think to start with would be good to remove all occurrences of service time get calls and try to work around it. Honestly I havent thought of a way around it either, going to try the pending event method.

**Comment by theshaydays** [comment]
@pearlynliu @bl7447 yea i wish there was a workaround but seems like pending is the way to go (even though i don't personally like it). Thanks a lot!!
